### PR TITLE
[6.0][Distributed] Adjust whenLocal impl to avoid potential ABI issues

### PR DESCRIPTION
**Pull request by ktoso**
**Description**: This is a follow up to https://github.com/swiftlang/swift/pull/74324 resolving rdar://129651474 where we added a missing `@AEIC` to the `whenLocal` function which adopted typed throws. This caused Distributed breakage in Beta 1. This PR here is a follow up to make the transition even safer, by introducing the "accidentally exposed symbol" (by not having made the new method AEIC from the beginning), such that applications which build against that we don't "remove" this symbol without replacement -- i.e. this patch doesn't "add" an API, but "keeps it" for ABI reasons, even if no new code against the new SDK should ever be touching that old symbol.

**Scope/Impact**: Distributed module users are impacted if they happened to build against an SDK with the non-`@AEIC` declaration of the whenLocal function.

**Risk:** Low, this only adds an "old" abi entrypoint which may have been used by someone in the brief beta 1 period.

To put it differently, this is "**strictly safer**" than the existing version.

**Testing**: CI testing and we verified manually with @xedin in a real reproducer project the AEIC approach is correct overall, and this only adds an "old" abi entrypoint.
**Reviewed by**: @xedin 

**Original PR:** https://github.com/swiftlang/swift/pull/74606
**Radar:** rdar://130610949

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-ci please test

**Review by tbkka — APPROVED** [review]
